libs/langchain/langchain/indexes/_api.py

```diff
@@ -330,7 +330,7 @@ def index(
         # Be pessimistic and assume that all vector store write will fail.
         # First write to vector store
         if docs_to_index:
-            vector_store.add_documents(docs_to_index, ids=uids)
+            vector_store.add_documents(docs_to_index, ids=uids, batch_size=batch_size)
```

Comment:
isn't `docs_to_index` already batched, as this logic is inside the `for doc_batch in _batch(batch_size, doc_iterator):` loop?

Reply:
Yes, but the batch size is not necessarily respected by the underlying vector store in the situation where the vector store's default batch size is less than the `batch_size` specified in the call to `index()`.

For example, if the vector store's default batch size is 64, but I call `index(..., batch_size=100)`, then the loop is batching in groups of 100, but sending to my vector store in groups of 64. I would expect that when I say "batch size is 100", I send 100 documents to my vector store per call (assuming these 100 documents have never been seen by the record manager).